lib/Secrets/lockcoderequest.h

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2018 Jolla Ltd.
+ * Contact: Chris Adams <chris.adams@jollamobile.com>
+ * All rights reserved.
+ * BSD 3-Clause License, see LICENSE.
+ */
+
+#ifndef LIBSAILFISHSECRETS_LOCKCODEREQUEST_H
+#define LIBSAILFISHSECRETS_LOCKCODEREQUEST_H
+
+#include "Secrets/secretsglobal.h"
+#include "Secrets/request.h"
+#include "Secrets/secret.h"
+#include "Secrets/secretmanager.h"
+
+#include <QtCore/QObject>
+#include <QtCore/QScopedPointer>
+#include <QtCore/QString>
+
+namespace Sailfish {
+
+namespace Secrets {
+
+class LockCodeRequestPrivate;
+class SAILFISH_SECRETS_API LockCodeRequest : public Sailfish::Secrets::Request
+{
+    Q_OBJECT
+    Q_PROPERTY(LockCodeRequestType lockCodeRequestType READ lockCodeRequestType WRITE setLockCodeRequestType NOTIFY lockCodeRequestTypeChanged)
+    Q_PROPERTY(QString secretName READ secretName WRITE setSecretName NOTIFY secretNameChanged)
+    Q_PROPERTY(QString collectionName READ collectionName WRITE setCollectionName NOTIFY collectionNameChanged)
+    Q_PROPERTY(Sailfish::Secrets::SecretManager::UserInteractionMode userInteractionMode READ userInteractionMode WRITE setUserInteractionMode NOTIFY userInteractionModeChanged)
+    Q_PROPERTY(Sailfish::Secrets::InteractionParameters interactionParameters READ interactionParameters WRITE setInteractionParameters NOTIFY interactionParametersChanged)
+
+public:
+    enum LockCodeRequestType {
+        ModifyLockCode = 0,
+        ProvideLockCode,
+        ForgetLockCode
+    };
+    Q_ENUM(LockCodeRequestType)
+
+    LockCodeRequest(QObject *parent = Q_NULLPTR);
+    ~LockCodeRequest();
```

Comment:
Should the destructor be `virtual` here? I would either suggest a `virtual` destructor, or marking the class as `final`.

Reply:
All of the Request types derive from Request base class which has a virtual dtor.